### PR TITLE
 Deprecate PILLOW_VERSION and VERSION

### PR DESCRIPTION
**Pull request by hugovk**
For #3082.

Replaces PR https://github.com/python-pillow/Pillow/pull/3085.

Changes proposed in this pull request:

 * Internally, use more conventional `__version__` rather than `PILLOW_VERSION`
 * Mark `PILLOW_VERSION` and `VERSION` as deprecated, we'll remove them in the next major version. 
   * It's been nearly a decade since the fork, and no end users will need to know the old PIL 1.1.7. It just causes confusion.
   * With that gone, no need for `PILLOW_VERSION`, use `__version__` instead.

---

https://github.com/python-pillow/Pillow/pull/3083#issuecomment-380353788:

> Don't break the single version in c & python,

Check

> don't break the world by removing the PIL 1.1.7 version until we do a major release

Check

### CODE REVIEW

File: src/PIL/__init__.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
 
 from . import version
 
+# PILLOW_VERSION and VERSION are deprecated and will be removed in Pillow 6.0.0. Use __version__ instead.
 VERSION = '1.1.7'  # PIL Version
 PILLOW_VERSION = version.__version__
```
Comment (homm) [inline src/PIL/__init__.py]: Could you rebase on the current master to see the full picture?

Comment (hugovk) [reply src/PIL/__init__.py]: Rebased!

File: src/PIL/ImageCms.py [hunk 2]
```diff
@@ -951,5 +951,5 @@ def versions():
 
     return (
         VERSION, core.littlecms_version,
-        sys.version.split()[0], Image.VERSION
+        sys.version.split()[0], Image.__version__
```
Comment (homm) [inline src/PIL/ImageCms.py]: It used to be 1.1.7 here.

Comment (hugovk) [reply src/PIL/ImageCms.py]: Yes, reverted.

File: docs/releasenotes/5.2.0.rst [hunk 3]
```diff
@@ -0,0 +1,22 @@
+5.2.0
+-----
+
+API Changes
+===========
+
+Deprecations
+^^^^^^^^^^^^
+
+Two version constants – ``VERSION`` (the old PIL version 1.1.7) and
```
Comment (homm) [inline docs/releasenotes/5.2.0.rst]: ``PIL.VERSION`` and so on

Comment (hugovk) [reply docs/releasenotes/5.2.0.rst]: Added.

**Comment by wiredfool** [comment]
You know, there's one other thing. 

When we ask for a version number, we should be able to ask for one thing. And unfortunately, that's going to be PILLOW_VERSION for a good long time. 

Otherwise, we're going to get this conversation:

>>> What version are you running?

>> How do I do that?

> Depends on what version you're running, either PIL.PILLOW_VERSION or PIL.\_\_version\_\_

**Comment by aclark4life** [comment]
That conversation is probably OK because that’s what deprecation is for. PILLOW_VERSION was a hack we should probably fix…

**Comment by hugovk** [comment]
PIL.__version__ has already been around for 2 years since 3.4.0, in 7 quarterly releases so far. 

(Commit https://github.com/python-pillow/Pillow/commit/7bfb2527bb0b832f830f87c37bd165028118e442 in PR https://github.com/python-pillow/Pillow/pull/2027.) 

So I think we're fine asking for PIL.__version__.

File: docs/releasenotes/5.2.0.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+These version constants have been deprecated and will be removed in the next
+major release:
+
+* ``PIL.VERSION`` (old PIL version 1.1.7)
+* ``PIL.PILLOW_VERSION``
+* ``PIL.Image.VERSION``
+* ``PIL.Image.PILLOW_VERSION``
+
+Use ``__version__`` instead.
```
Comment (homm) [inline docs/releasenotes/5.2.0.rst]: PIL.__version__ ?

Comment (hugovk) [reply docs/releasenotes/5.2.0.rst]: Good idea, updated.

**Comment by homm** [comment]
You know, I'm for the depreciating VERSION and leaving PILLOW_VERSION as is. I don't think this is so terrible to have fallback for code which works from the beging of the library. Anyway, we already reduced the number of "version" constants from four to two in next release.

**Comment by homm** [comment]
@hugovk Needs rebase

**Comment by aclark4life** [comment]
@hugovk Sounds good… I think saying we're going to get rid of ``PILLOW_VERSION`` at some point but not specifying _which_ future version will allow folks plenty of time to comment if they care. As the person who probably added ``PILLOW_VERSION``, I think it should eventually go…

**Comment by hugovk** [comment]
Rebased and updated.

File: docs/releasenotes/5.2.0.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+-----
+
+API Changes
+===========
+
+Deprecations
+^^^^^^^^^^^^
+
+These version constants have been deprecated. ``VERSION`` will be removed in
+Pillow 6.0.0, and ``PILLOW_VERSION`` will be removed in a future release.
```
Comment (homm) [inline docs/releasenotes/5.2.0.rst]: Doesn't it mean that PILLOW_VERSION will be removed in a future release i.e. in 5.3, before 6.0? For me, "and ``PILLOW_VERSION`` will be removed after that." looks more clear.

Comment (hugovk) [reply docs/releasenotes/5.2.0.rst]: It is ambiguous. Updated!

**Comment by jwilk** [comment]
I was using ``PILLOW_VERSION`` to [distinguish between the original PIL and Pillow](https://github.com/jwilk/scanhelper/blob/ecad238aae9e00eee4cc867cb28dc8006e310ec6/lib/cli.py#L92).
Is there another way to distinguish between the two?

**Comment by radarhere** [comment]
PIL doesn't have `__version__`, so you can use that as a direct continuation of your current check

**Pull request merged**